src/domain/usecases/data-entry/DownloadTemplateUseCase.ts

```diff
@@ -0,0 +1,197 @@
+import { SheetBuilder } from "../../../data/repositories/download-template/sheetBuilder";
+import { ExcelBuilder } from "../../helpers/ExcelBuilder";
+import { getTemplateId } from "../../../data/repositories/ExcelPopulateDefaultRepository";
+import * as templates from "../../entities/data-entry/program-templates";
+import { EGASPProgramDefaultRepository } from "../../../data/repositories/download-template/EGASPProgramDefaultRepository";
+import { EGASP_PROGRAM_ID } from "../../../data/repositories/program-rule/ProgramRulesMetadataDefaultRepository";
+import {
+    AMC_PRODUCT_REGISTER_PROGRAM_ID,
+    AMC_RAW_PRODUCT_CONSUMPTION_CALCULATED_STAGE_ID,
+    AMC_RAW_PRODUCT_CONSUMPTION_STAGE_ID,
+} from "./amc/ImportAMCProductLevelData";
+import {
+    AMC_RAW_SUBSTANCE_CONSUMPTION_PROGRAM_ID,
+    AMC_SUBSTANCE_CALCULATED_CONSUMPTION_PROGRAM_ID,
+} from "./amc/ImportAMCSubstanceLevelData";
+import { MetadataRepository } from "../../repositories/MetadataRepository";
+
+export type FileType = "PRODUCT" | "SUBSTANCE";
+export type DownloadType = "SUBMITTED" | "CALCULATED";
+
+export interface DownloadTemplateProps {
+    moduleName: string;
+    fileType: "PRODUCT" | "SUBSTANCE";
+    downloadType?: "SUBMITTED" | "CALCULATED";
+    orgUnit: string;
+    startDate?: Moment;
+    endDate?: Moment;
+    populate?: boolean;
+    populateStartDate?: Moment;
+    populateEndDate?: Moment;
+    downloadRelationships: boolean;
+    filterTEIEnrollmentDate?: boolean;
+    relationshipsOuFilter?: RelationshipOrgUnitFilter;
+    templateId?: string;
+    templateType?: TemplateType;
+    splitDataEntryTabsBySection?: boolean;
+    useCodesForMetadata: boolean;
+}
+
+export class DownloadTemplateUseCase implements UseCase {
+    constructor(
+        private DownloadtemplateRepository: DownloadTemplateRepository,
+        private excelRepository: ExcelRepository,
+        private egaspRepository: EGASPProgramDefaultRepository,
+        private metadataRepository: MetadataRepository
+    ) {}
+
+    public async execute({
+        moduleName,
+        fileType,
+        downloadType,
+        orgUnit,
+        startDate,
+        endDate,
+        populate,
+        populateStartDate,
+        populateEndDate,
+        downloadRelationships = true,
+        filterTEIEnrollmentDate,
+        relationshipsOuFilter,
+        splitDataEntryTabsBySection = false,
+        useCodesForMetadata = false,
+    }: DownloadTemplateProps): Promise<File> {
+        const { programId, programStageId } = getProgramId(moduleName, fileType, downloadType);
+        const formType = getFormType(programId);
+        const settings = await this.egaspRepository.getTemplateSettings().toPromise();
+        const template = this.getTemplate(programId);
+        if (!template) {
+            throw new Error("No template found for this Program");
+        }
+
+        const element = await this.DownloadtemplateRepository.getElement(formType, programId);
+
+        let orgUnits = [orgUnit];
```

Comment:
do not use let.  
const orgUnits = moduleName === "EGASP" ? clinicsAndLabsOrgUnits : [orgUnit]